### PR TITLE
WT-7573 Print an error message and exit for invalid backup configurations in wtperf tests.

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: bench/wtperf/config.c [hunk 1]
```diff
@@ -778,13 +778,20 @@ config_sanity(WTPERF *wtperf)
...
+        fprintf(stderr, "Direct I/O cannot be used when backup is configured.\n");
+        return (EINVAL);
+    }
+
     if (wtperf->workload != NULL)
         for (i = 0, workp = wtperf->workload; i < wtperf->workload_cnt; ++i, ++workp) {
             if (opts->readonly &&
               (workp->insert != 0 || workp->modify != 0 || workp->truncate != 0 ||
-                workp->update != 0)) {
+                workp->update != 0 || opts->backup_interval != 0)) {
```
Comment (etienneptl) [inline bench/wtperf/config.c:791]: Should we keep backup related checks seperate as I did line 782 ?

Comment (etienneptl) [reply bench/wtperf/config.c:791]: I think it is better as it is not related to wtperf->workload, I will proceed.

File: bench/wtperf/config.c [hunk 2]
```diff
@@ -778,6 +778,13 @@ config_sanity(WTPERF *wtperf)
             opts->value_sz_min = opts->value_sz;
     }
 
+    /* When backup is configured, direct I/O and readonly cannot be used. */
+    if (opts->backup_interval != 0)
```
Comment (jiechenbo) [inline bench/wtperf/config.c:782]: Alphabetize this, with the other if conditions

File: bench/wtperf/config.c [hunk 3]
```diff
@@ -778,6 +778,13 @@ config_sanity(WTPERF *wtperf)
             opts->value_sz_min = opts->value_sz;
     }
 
+    /* When backup is configured, direct I/O and readonly cannot be used. */
```
Comment (jiechenbo) [inline bench/wtperf/config.c:781]: I think we should include why direct I/O or readonly can't be used.

**Review by jiechenbo — CHANGES_REQUESTED** [review]

**Review by jiechenbo — APPROVED** [review]
LGTM. Nice work!

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**